src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -196,14 +200,33 @@ private UpdateDocumentsCommand prepareRequestForAttachingDocumentToCase(CaseDocu
         List<DocumentUpdate> documentsList = new ArrayList<>();
         for (DocumentHashToken documentHashToken : caseDocumentsMetadata.getDocumentHashTokens()) {
 
+            Optional<StoredDocumentHalResource> documentMetadata =
+                getDocumentMetadata(UUID.fromString(documentHashToken.getId()));
+
             if (documentHashToken.getHashToken() != null) {
```

Comment:
May be extract this whole if logic into a separate private method . `if (documentHashToken.getHashToken() != null)`